CONTRIBUTING.md

```diff
@@ -0,0 +1,67 @@
+# Contributing
+
+Thanks a lot for your interest in contributing to **editorconfig-checker**!
+
+## Types of contributions
+
+Contributions to **editorconfig-checker** _include_, but is _not limited to_:
+
+- Report a bug.
+- Suggest a new feature idea.
+- Correct spelling errors, improvements, or additions to documentation files (README, CONTRIBUTING...).
+- Improve structure/format/performance/refactor/tests of the code.
+
+All contributions are welcome from anyone willing to work in good faith with other contributors and the community. No contribution is too small and all contributions are valued.
+
+## Open Development
+
+All work on **editorconfig-checker** happens directly on [GitHub](https://github.com/editorconfig-checker/editorconfig-checker). Both core team members and external contributors send pull requests, which go through the same review process.
+
+## Pull Requests
+
+Pull Requests are the way concrete changes are made to the code and the documentation.
+
+**Please first discuss** the change you wish to make via [issue](https://github.com/editorconfig-checker/editorconfig-checker/issues) before making a change. It might save you time.
+
+If you're adding new features to **editorconfig-checker**, please include tests.
+
+### Review process and requirements to merge
+
+To ensure everyone in the community has the opportunity to share their feedback, ideas, and suggestions, a Pull Request must go through a review process, and must meet the following requirements to be merged:
+
+- **Approval:** At least **2 core team members** must approve the Pull Request.
+- **Open Duration:** The Pull Request must be **open for at least 72 hours**.
+- **CI checks:** The Pull Request must **pass all CI checks**, including tests and build.
+
+**Note:** If a critical bug fix is required, and the Pull Request was approved by at least **3 core team members**, the Pull Request **can be merged before the 72 hours** open duration.
+
+Pull requests are merged using the `Squash and Merge` option to keep the commit history clean, and promote clear separation of changes in separated pull requests.
```

Comment:
This is off topic for a contributing guide because most contributors will not be merging their PR.

I'm also not a fan of having a blanket "always squash" policy for project maintainers. When a pull request contains non-atomic "fixup" commits, they should definitely be squashed before merging. However, a contributor may intentionally put multiple atomic commits into a pull request and in this case squashing the commits is harmful.

Reply:
I agree. Talking about the process of merging is a bit out of contributions guide.

But at same time, having a clear policy is good for maintainers.

But that said, I also disagree about the fact every PR must be squashed. Sometimes separate commits are interesting.

So maybe the bguide could talk about the fact, that most PRs will be squashed.

Reply:
With the current settings on GitHub of this repo, PRs can only be squashed, it is not possible to merge them with multiple commits (however, this can of course change).

Also squash-and-merge doesn't erase history, as it will still be visible in the PR, and as said promotes clear separation of changes into multiple PRs, and therefore makes it easier to review.

Reply:
Resolved by [`923493b` (#378)](https://github.com/editorconfig-checker/editorconfig-checker/pull/378/commits/923493be7d65693b0d2cb71122ef95957da53589)